src/sparse_voxels.jl

```diff
@@ -79,24 +76,25 @@ function SparseVoxelGrid{T <: AbstractVector}(points::Vector{T}, voxel_size)
     return SparseVoxelGrid(voxel_size, voxel_info, point_indices)
 end
 
-# Convert matrix to Vector{SVector}
-function SparseVoxelGrid{T <: Number}(points::Matrix{T}, voxel_size)
-    ndim = size(points, 1)
+function SparseVoxelGrid{T1 <: Real, T2 <: Real}(points::Matrix{T1}, voxel_size::SVector{3, T2})
     npoints = size(points, 2)
-    if isbits(T)
-        new_data = reinterpret(SVector{ndim, T}, points, (length(points) ÷ ndim, ))
-    else
-        new_data = SVector{ndim, T}[SVector{ndim, T}(points[:, i]) for i in 1:npoints]
-    end
+    new_data = Vector{SVector{3, T1}}()
+    new_data = reinterpret(SVector{3, T1}, points, (length(points) ÷ 3, ))
     SparseVoxelGrid(new_data, voxel_size)
 end
 
-function get_voxel_size{T <: Number}(voxel_size::T)
-    SVector{3, T}(voxel_size, voxel_size, voxel_size)
+function SparseVoxelGrid{T1 <: Real, T2 <: Real}(points::Matrix{T1}, voxel_size::T2)
+    voxel_size_vector = get_voxel_size(voxel_size)
+    SparseVoxelGrid(points, voxel_size_vector)
 end
 
-function get_voxel_size{T <: Number}(voxel_size::NTuple{3, T})
-    SVector{3, T}(voxel_size)
+function SparseVoxelGrid{T1 <: AbstractVector, T2 <: Real}(points::Vector{T1}, voxel_size::T2)
+    voxel_size_vector = get_voxel_size(voxel_size)
+    SparseVoxelGrid(points, voxel_size_vector)
+end
+
+function get_voxel_size{T <: Real}(voxel_size::T)
```

Comment:
If we have `get_voxel_size(s::SVector) = s` then we can remove the overloads for `SparseVoxelGrid` above.